backend/routes/seatbid.js

```diff
@@ -0,0 +1,186 @@
+    const expirationTime = Date.now() + 30 * 60 * 1000; // 30 minutes
+
+    const existingBidding = Array.from(activeBiddings.values()).find(
+        bidding => bidding.passengerId === passengerId && bidding.status === 'ACTIVE'
+    );
+
+    if (existingBidding) {
+        return res.status(400).json({ error: 'You already have an active bidding' });
+    }
+
+    const newBidding = {
+        passengerId,
+        seatNumber,
+        flightId,
+        startTime: Date.now(),
+        expirationTime,
+        status: 'ACTIVE'
+    };
+
+    activeBiddings.set(biddingId, newBidding);
+    bids.set(biddingId, []);
+
+    emitToFlight(req.app.locals.io, flightId, 'new-bid', { 
+        biddingId, 
+        seatNumber, 
+        passengerId,
+        bids: []
+    });
+
+    setTimeout(() => expireBidding(req.app.locals.io, biddingId), expirationTime - Date.now());
+
+    res.json({ success: true, biddingId, ...newBidding, bids: [] });
+});
+
+/**
+ * @route POST /bid
+ * @desc Place a bid on an active bidding
+ * @access Private
+ */
+router.post('/bid', authenticateUser, async (req, res) => {
+    const { biddingId, bidderId, amount, bidderSeatNumber } = req.body;
+    const bidding = activeBiddings.get(biddingId);
+
+    if (!bidding || bidding.status !== 'ACTIVE') {
+        return res.status(400).json({ error: 'Bidding not found or not active' });
+    }
+
+    const bidderActiveBidding = Array.from(activeBiddings.values()).find(
+        b => b.passengerId === bidderId && b.status === 'ACTIVE'
+    );
+
+    if (bidderActiveBidding) {
+        bidderActiveBidding.status = 'CANCELLED';
+        emitToFlight(req.app.locals.io, bidderActiveBidding.flightId, 'bidding-cancelled', { biddingId: bidderActiveBidding.biddingId });
+    }
+
+    const bid = { 
+        id: Date.now().toString(), 
+        bidderId, 
+        amount, 
+        bidderSeatNumber, 
+        time: Date.now() 
+    };
+
+    if (!bids.has(biddingId)) bids.set(biddingId, []);
+    bids.get(biddingId).push(bid);
+
+    emitToFlight(req.app.locals.io, bidding.flightId, 'new-bid', { biddingId, bid });
+
+    res.json({ success: true, bidId: bid.id });
+});
+
+/**
+ * @route POST /accept
+ * @desc Accept a bid and complete the seat swap
+ * @access Private
+ */
+router.post('/accept', authenticateUser, async (req, res) => {
+    const { biddingId, bidId } = req.body;
+    const bidding = activeBiddings.get(biddingId);
+
+    if (!bidding || bidding.status !== 'ACTIVE') {
+        return res.status(400).json({ error: 'Bidding not found or not active' });
+    }
+
+    const acceptedBid = bids.get(biddingId).find(bid => bid.id === bidId);
+    if (!acceptedBid) {
+        return res.status(400).json({ error: 'Bid not found' });
+    }
+
+    try {
+        await prisma.$transaction([
+            prisma.passenger.update({
+                where: { flightId_userId: { flightId: bidding.flightId, userId: bidding.passengerId } },
+                data: { seatNumber: 'TEMP' },
```

Comment:
what's this? we do we need to assign 'TEMP' and final seat in same transaction?

Reply:
My database has unique constraint on seat number and I can't directly swap seats which violated the unique constraint momentarily. So I used the 'TEMP' seat assignment as an intermediate step in the transaction to temporarily free up the initiator's seat. It allows a clean swap of seat numbers between two passengers without violating the unique constraint on seat assignments for a given flight.